### ISSUE TITLE
[BUG] Kirby-header not allowing nested components

### ISSUE THREAD

**Issue by NillerW** [kirbydesign/designsystem#3217]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
Kirby-header provides the functionality of adding a Kirby-flag, which is not working when the Kirby-flag is nested in a wrapper component. 

The same issue applies for avatar.

## Describe how to reproduce the bug
Create a new component which only holds a Kirby-flag and place it in the Kirby-header where the Kirby-flag should have been placed.

## Which Kirby version was used?
8.11.1 

## What was the expected behavior?
Kirby-header should display a nested Kirby-flag as if it was placed directly without the wrapper component.

## Please complete the following information:
- OS: MacOS
- Browser: Chrome
- Browser version: 116.0.5845.140 (Official Build) (arm64)

## Are there any additional context?
The problem resides in the header.component, and is two-fold. 

1) The header component has a ng-content that selects a Kirby-flag. 
2) The header component has a contentChild for KirbyFlagComponent, and a check for the child component being present in the DOM.

Issue 1 could be bypassed by using ngProjectAs on the nested flag component. However ngProjectAs does not work on contentChild, so due to the check for the content child being present, the nested component is still not shown.

This issue has been discussed with @jakobe.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/main/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by RasmusKjeldgaard** [comment]
Closing this issue automatically because it has been open for more than 6 months and is not currently in progress. If this is still relevant, please feel free to reopen it or open a new issue with updated details. Thank you!
